cmd/tikv-ctl/src/cmd.rs

```diff
@@ -627,6 +630,62 @@ pub enum Cmd {
         /// hex end key
         end: String,
     },
+    CompactLogBackup {
+        #[structopt(
+            short,
+            long,
+            default_value = "compaction",
+            help(
+                "name of the compaction, register this will help you find the compaction easier."
+            )
+        )]
+        name: String,
+        #[structopt(long = "from", help("from when we need to compact."))]
+        from_ts: u64,
+        #[structopt(
+            long = "until",
+            help(
+                "until when we need to compact. \
+                Also note that records out of the [--from, --until) range may also be compacted \
+                if their neighbour in the same file needs to be compacted."
+            )
+        )]
+        until_ts: u64,
+        #[structopt(
+            short = "N",
+            long = "concurrency",
+            default_value = "32",
+            help("how many compactions can be executed concurrently.")
+        )]
+        max_compaction_num: u64,
+        #[structopt(
+            short = "s",
+            long = "storage-base64",
+            help(
+                "the base-64 encoded protocol buffer message `StorageBackend`. \
+                `br` CLI should provide a subcommand that converts an URL to it."
+            )
+        )]
+        storage_base64: String,
+        #[structopt(
+            long,
+            default_value = "lz4",
+            help(
+                "the compression method will use when generating SSTs. (hint: zstd | lz4 | snappy)"
+            )
+        )]
+        compression: SstCompressionType,
+        #[structopt(
+            long,
+            help(
+                "the compression level. it definition and effect varies by the algorithm we choose."
+            )
+        )]
+        compression_level: Option<i32>,
+
+        #[structopt(long, help("Don't try to skip already finished compactions."))]
```

Comment:
what would happen if re-gen on finished compactions?

Reply:
It will be regenerated.

Reply:
So, it's an expected behavior? what does this `don't try to skip` mean here?

Reply:
That means skipping checkpoints.